KIPs/kip-81.md

```diff
@@ -0,0 +1,335 @@
+
+| Name              | Meaning                                                                                   | Value     |
+|----------------   |----------------------------------------------------------------------------------------   |--------   |
+| `votingDelay`     | Delay from proposal submission to voting start. Equals to the length of Pending state.    | 7 days    |
+| `votingPeriod`    | Duration of the voting. Equals the length of Active state.                                | 7 days    |
+| `queueTimeout`    | Grace period given to executors for queue() passed proposals.                             | 7 days    |
+| `execDelay`       | A minimum delay before a queued transaction can be executed.                              | 2 days    |
+| `execTimeout`     | Grace period given to executors for execute() queued proposals after `execDelay`.         | 7 days    |
+
+For each proposal, several important block numbers are determined.
+
+| Name              | Meaning                                                                                                                                                                   |
+|-----------------  |-------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `proposedBlock`   | The block number where `propose()` was called.                                                                                                                            |
+| `startBlock`      | Voting start block `proposedBlock + votingDelay`.                                                                                                                         |
+| `endBlock`        | Voting end block `startBlock + votingPeriod`. An executor should queue transactions in the proposal before `endBlock + queueTimeout`, otherwise the proposal expires.     |
+| `queuedBlock`     | The block number where `queue()` was called. An executor should trigger the transaction after `queuedBlock + execDelay`, before `queuedBlock + execDelay + execTimeout`.  |
+
+The timings are illustrated in below diagram.
+
+![proposal timing](../assets/kip-81/proposal_timing.png)
+
+A proposal is in one of the following states.
+
+- **Pending**: Proposer submitted the proposal but voting has not started. The pending state lasts for `votingDelay`. This state is also called the **notice period**.
+- **Active**: Ongoing voting. Voters can cast their votes. The active state lasts for `votingPeriod`. This state is also called the **voting period**.
+- **Canceled**: Proposer has canceled the proposal before execution. Any further interaction is prohibited.
+- **Passed**: Voting finalized and quorum reached. An executor shall queue the actions before a set timeout of `queueTimeout`. If the proposal contains no actions, any further interaction is prohibited. Otherwise, the proposal can proceed to Queued and Executed states.
```

Comment:
```suggestion
- **Passed**: Voting finalized and a quorum reached. An executor shall queue the actions before a set timeout of `queueTimeout`. If the proposal contains no actions, any further interaction is prohibited. Otherwise, the proposal can proceed to Queued and Executed states.
```